test/quantization/fx/test_model_report_fx.py

```diff
@@ -1319,5 +1304,92 @@ def test_input_weight_equalization_report_gen_empty(self):
+        with override_quantized_engine('fbgemm'):
+
+            detector_set = set([OutlierDetector(reference_percentile=0.95)])
+            model_report = ModelReport(detector_set)
+
+            # get tst model and callibrate
+            non_fused = self._get_prepped_for_calibration_model(self.LargeBatchModel(), detector_set)
+            fused = self._get_prepped_for_calibration_model(self.LargeBatchModel(), detector_set)
+
+            # reporter should still give same counts even for fused model
+            for prepared_for_callibrate_model, mod_report in [non_fused, fused]:
+
+                # supported modules to check
+                mods_to_check = set([nn.Linear, nn.Conv2d, nn.ReLU])
+
+                # get the set of all nodes in the graph their fqns
+                node_fqns = set([node.target for node in prepared_for_callibrate_model.graph.nodes])
+
+                # there should be 3 node fqns that have the observer inserted
+                correct_number_of_obs_inserted = 3
+                number_of_obs_found = 0
+                obs_name_to_find = InputWeightEqualizationDetector.DEFAULT_PRE_OBSERVER_NAME
+
+                for node in prepared_for_callibrate_model.graph.nodes:
+                    # if the obs name is inside the target, we found an observer
+                    if obs_name_to_find in str(node.target):
+                        number_of_obs_found += 1
+
+                self.assertEqual(number_of_obs_found, correct_number_of_obs_inserted)
+
+                # assert that each of the desired modules have the observers inserted
+                for fqn, module in prepared_for_callibrate_model.named_modules():
+                    # check if module is a supported module
+                    is_in_include_list = sum(list(map(lambda x: isinstance(module, x), mods_to_check))) > 0
+
+                    if is_in_include_list:
+                        # make sure it has the observer attribute
+                        self.assertTrue(hasattr(module, InputWeightEqualizationDetector.DEFAULT_PRE_OBSERVER_NAME))
+                    else:
+                        # if it's not a supported type, it shouldn't have observer attached
+                        self.assertTrue(not hasattr(module, InputWeightEqualizationDetector.DEFAULT_PRE_OBSERVER_NAME))
+
+
+def _get_prepped_for_calibration_model_helper(model, detector_set, example_input, fused: bool = False):
+    r"""Returns a model that has been prepared for callibration and corresponding model_report"""
+    # set the backend for this test
+    torch.backends.quantized.engine = "fbgemm"
+
+    model_report = ModelReport(detector_set)
+
+    # create model instance and prepare it
+    example_input = example_input.to(torch.float)
+    q_config_mapping = torch.ao.quantization.get_default_qconfig_mapping()
+
+    # if they passed in fusion paramter, make sure to test that
+    if fused:
+        model = torch.quantization.fuse_modules(model, [['conv', 'relu']])
```

Comment:
this does not look like a general helper function, should we attach this to the model definition (LargeBatchModel)?

Reply:
it is general as it is used for multiple different models across multiple different test classes. For example, lines 1153 and 1154 (in `test_input_weight_equalization_determine_points`) also make use of this to test a fusion vs. non-fusion case for a different model class.